src/Collection/CollectionTrait.php

```diff
@@ -316,10 +316,9 @@ public function first()
      */
     public function append($items)
     {
-        $items = $items instanceof Iterator ? $items : new Collection($items);
         $list = new AppendIterator;
         $list->append($this);
```

Comment:
sure, send it as a pull request if you like. I can do it on my evening otherwise :)